internal/pkg/warewulfd/api.go

```diff
@@ -0,0 +1,106 @@
+package warewulfd
+
+import (
+	"context"
+	"errors"
+	"fmt"
+
+	"github.com/swaggest/openapi-go/openapi3"
+	"github.com/swaggest/rest/web"
+	swgui "github.com/swaggest/swgui/v5emb"
+	"github.com/swaggest/usecase"
+	"github.com/swaggest/usecase/status"
+
+	"github.com/warewulf/warewulf/internal/pkg/node"
+	"github.com/warewulf/warewulf/internal/pkg/version"
+)
+
+func apiHandler() *web.Service {
+	api := web.NewService(openapi3.NewReflector())
+	if r3, ok := api.OpenAPIReflector().(*openapi3.Reflector); ok && api.OpenAPI == nil {
+		api.OpenAPI = r3.Spec
+	}
+
+	api.OpenAPISchema().SetTitle("Warewulf v4 API")
+	api.OpenAPISchema().SetDescription("This service provides an API to a Warewulf v4 server.")
+	api.OpenAPISchema().SetVersion(version.GetVersion())
+
+	api.Get("/api/nodes", apiGetNodes())
+	api.Get("/api/nodes/{id}", apiGetNodeByID())
+	api.Get("/api/profiles", apiGetProfiles())
+	api.Get("/api/profiles/{id}", apiGetProfileByID())
+
+	api.Docs("/api/docs", swgui.New)
+
+	return api
+}
+
```

Comment:
So this will only output the unmerged nodes. I strongly disagree deliver the nodes like this in the API, as this is just a complicated way to transform `nodes.conf` to yaml. But this is just a draft, so please provide the merged nodes.

Reply:
Yep, definitely just a draft. I'm not looking for a review at this time; I'm pushing it so people can see, but I'm expecting heavy iteration while I prototype.

The reason I started with the unmerged nodes is because that's what people would _push_. I currently intend to switch the default view to the merged node and then support a separate `/api/raw-nodes/` or similar; but I'm still playing with it in my head.

Reply:
The first thing I would like to see in an API would be the possibility to push a status string to `warewulfd` which then is displayed with `wwctl status`.